### PR TITLE
Warn about NGINX LB being recreated with upgrade to 1.8.0+

### PR DESCRIPTION
**Pull request by sslavic**
Towards https://github.com/giantswarm/giantswarm/issues/12881

### CODE REVIEW

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+### :zap: Dear Giant Swarm customer :zap:
+
+We'd like to inform you about a known issue when upgrading NGINX ingress controller (IC) optional app to v1.8.0+ on AWS and Azure clusters - NGINX IC LoadBalancer Service may get renamed/recreated with the app upgrade.
+
+In older releases the Service name was hardcoded to `nginx-ingress-controller`. As of v1.8.0, to ensure its uniqueness for multiple NGINX ICs per cluster support, the LoadBalancer Service name was made to be dynamic, derived from Helm release i.e. App Custom Resource name.
```
Comment (sslavic) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md:5]:
```suggestion
In older releases the Service name was hardcoded to `nginx-ingress-controller`. As of v1.8.0, to ensure its uniqueness for multiple NGINX ICs per cluster support, the LoadBalancer Service name was made to be dynamic, derived from Helm release i.e. App Custom Resource (CR) name.
```

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+### :zap: Dear Giant Swarm customer :zap:
+
+We'd like to inform you about a known issue when upgrading NGINX ingress controller (IC) optional app to v1.8.0+ on AWS and Azure clusters - NGINX IC LoadBalancer Service may get renamed/recreated with the app upgrade.
+
+In older releases the Service name was hardcoded to `nginx-ingress-controller`. As of v1.8.0, to ensure its uniqueness for multiple NGINX ICs per cluster support, the LoadBalancer Service name was made to be dynamic, derived from Helm release i.e. App Custom Resource (CR) name.
+NGINX IC LoadBalancer Service gets replaced by a new one for every NGINC IC App CR whose name is not `nginx-ingress-controller`.
```
Comment (sslavic) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md:6]:
```suggestion
Therefore, NGINX IC LoadBalancer Service will be replaced by a new one for every NGINC IC App CR whose name is not `nginx-ingress-controller`.
```

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+### :zap: Dear Giant Swarm customer :zap:
+
+We'd like to inform you about a known issue when upgrading NGINX ingress controller (IC) optional app to v1.8.0+ on AWS and Azure clusters - NGINX IC LoadBalancer Service may get renamed/recreated with the app upgrade.
```
Comment (omegas27) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md:3]:
```suggestion
We'd like to inform you about a known issue when upgrading NGINX ingress controller (IC) optional app to v1.8.0+ on *AWS and Azure* clusters - NGINX IC LoadBalancer Service may get renamed/recreated with the app upgrade.
```

**Comment by omegas27** [comment]
This looks fine to me

**Review by omegas27 — APPROVED** [review]

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md (hunk 3 above)
Comment (sslavic) [reply announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md:3]: Fine with making it bold, still I assume this will be sent to AWS and Azure customers only.

Comment (omegas27) [reply announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md]: True, but some customers have both cloud and on prem

**Comment by sslavic** [comment]
Added the short version, for Slack message, with link to the long one. PTAL

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+### :zap: Dear Giant Swarm customer :zap:
```
Comment (cokiengchiara) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md:1]:
```suggestion
## Known Issue: Potential ingress downtime when upgrading to NGINX IC v1.8.0+
```

Make title more descriptive.

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+ - short.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+:information_source: There's a known issue, potential ingress downtime when upgrading optional NGINX IC App from older to v1.8.0+
+
+Please read the full notice on https://github.com/giantswarm/releases/tree/master/announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md
```
Comment (cokiengchiara) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+ - short.md:3]:
```suggestion
Please read the full notice on [Known Issue: Potential ingress downtime when upgrading to NGINX IC v1.8.0+](https://github.com/giantswarm/releases/tree/master/announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md)
```

Comment (cokiengchiara) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+ - short.md:3]:
```suggestion
Please read the full notice and how to minimize impact in https://github.com/giantswarm/releases/tree/master/announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md
```

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+When NGINX IC LoadBalancer Service gets recreated, cloud service provider (CSP) load balancer behind it gets recycled as well.
+It can take minute or so for ingress DNS records to be updated by external-dns and change propagated to clients.
+During that time there's ingress traffic downtime, since clients still resolve old no longer present CSP load balancer.
+
+Please take the potential downtime into consideration when planning the upgrade.
+
+To make sure the downtime is shortest possible, external-dns availability is important precondition.
+In recent platform releases (Azure v12.0.2, and AWS v12.1.4 and v11.5.4) we've improved external-dns monitoring and alerting.
+
+Therefore, before upgrading NGINX IC optional app to 1.8.0+, please make sure that your cluster has been upgraded to the latest release.
```
Comment (cokiengchiara) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md:17]:
```suggestion
**Therefore, before upgrading NGINX IC optional app to 1.8.0+, please make sure that your cluster has been upgraded to the latest release.**
```

Bold the actionable.

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+When NGINX IC LoadBalancer Service gets recreated, cloud service provider (CSP) load balancer behind it gets recycled as well.
+It can take minute or so for ingress DNS records to be updated by external-dns and change propagated to clients.
+During that time there's ingress traffic downtime, since clients still resolve old no longer present CSP load balancer.
+
+Please take the potential downtime into consideration when planning the upgrade.
+
+To make sure the downtime is shortest possible, external-dns availability is important precondition.
+In recent platform releases (Azure v12.0.2, and AWS v12.1.4 and v11.5.4) we've improved external-dns monitoring and alerting.
+
+**tl;dr: Therefore, before upgrading NGINX IC optional app to 1.8.0+, please make sure that your cluster has been upgraded to the latest release.**
```
Comment (sslavic) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md:17]:
```suggestion
**tl;dr: Therefore, before upgrading NGINX IC optional app to 1.8.0+, please make sure that your cluster has been upgraded to the latest platform release.**
```

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+We'd like to inform you about a known issue when upgrading NGINX ingress controller (IC) optional app to v1.8.0+ on **AWS** and **Azure** clusters - NGINX IC LoadBalancer Service may get renamed/recreated with the app upgrade.
+
+In older releases the Service name was hardcoded to `nginx-ingress-controller`. As of v1.8.0, to ensure its uniqueness for multiple NGINX ICs per cluster support, the LoadBalancer Service name was made to be dynamic, derived from Helm release i.e. App Custom Resource (CR) name.
+Therefore, NGINX IC LoadBalancer Service will be replaced by a new one for every NGINC IC App CR whose name is not `nginx-ingress-controller`.
+
+When NGINX IC LoadBalancer Service gets recreated, cloud service provider (CSP) load balancer behind it gets recycled as well.
+It can take minute or so for ingress DNS records to be updated by external-dns and change propagated to clients.
+During that time there's ingress traffic downtime, since clients still resolve old no longer present CSP load balancer.
+
+Please take the potential downtime into consideration when planning the upgrade.
```
Comment (sslavic) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md:12]:
```suggestion
** Please take the potential downtime into consideration when planning the upgrade. **
```

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+ - short.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+:information_source: There's a known issue, potential ingress downtime when upgrading optional NGINX IC App from older to v1.8.0+
+
+Please read the full notice on [Known Issue: Potential ingress downtime when upgrading to NGINX IC v1.8.0+](https://github.com/giantswarm/releases/tree/master/announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md)
```
Comment (sslavic) [inline announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+ - short.md:3]:
```suggestion
Please read the full notice and how to minimize impact in [Known Issue: Potential ingress downtime when upgrading to NGINX IC v1.8.0+](https://github.com/giantswarm/releases/tree/master/announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md)
```

File: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+ - short.md (hunk 5 above)
Comment (sslavic) [reply announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+ - short.md]: Multiple suggestions for same line - I merged one, but it invalidated this second one. I've reapplied it myself. PTAL

**Review by cokiengchiara — APPROVED** [review]
LGTM. Lets give it until 2:30pm in SIG Product.

**Review by ubergesundheit — APPROVED** [review]
LGTM
